<!--
Brings up the flirtlib place recognition node

Arguments:
- run_db: whether to run a Mongo server on localhost or use an existing
  one.  Default is false.
- db_path: db path.  Used if run_db is true.  Default is /removable/flirtlib_db
- remote_db_host: hostname of remote db.  Used if run_db is false.  Default
  is mongo.willowgarage.com

Requires:
- Nav stack to be running
- Laser scans on base_scan topic

Provides:
- Attempts to localize the robot on startup if mislocalization detected
- Saves new scans to the db once at least one successful navigation happens
-->

<launch>
  <arg name="run_db" default="false"/>
  <arg name="db_path" default="/removable/flirtlib_db"/>
  <arg name="remote_db_host" default="mongo.willowgarage.com"/>

  <arg name="db_host" value="$(arg remote_db_host)" unless="$(arg run_db)"/>
  <arg name="db_host" value="localhost" if="$(arg run_db)"/>

  <node pkg="flirtlib_ros" type="localization_monitor"
        name="flirtlib_place_rec" output="screen">
    <param name="min_successful_navs" value="1"/>
    <param name="min_num_matches" value="10"/>
    <param name="db_name" value="flirtlib_place_rec"/>
    <param name="db_host" value="$(arg db_host)"/>
  </node>

  <node pkg="mongodb" type="wrapper.py" name="flirtlib_db" if="$(arg run_db)"
        output="screen"/>
</launch>
